### PR TITLE
Remove dependency on mruby-symbol-ext

### PR DESCRIPTION
**Pull request by tkbky**
Part of #32 to remove dependency on mruby-*-ext mrbgems.

### CODE REVIEW

**Comment by tkbky** [comment]
The following C methods are not implemented.

```c
#ifdef MRB_ENABLE_ALL_SYMBOLS
  mrb_define_class_method(mrb, s, "all_symbols", mrb_sym_all_symbols, MRB_ARGS_NONE());
#endif
  mrb_define_method(mrb, s, "length", mrb_sym_length, MRB_ARGS_NONE());
  mrb_define_method(mrb, s, "size", mrb_sym_length, MRB_ARGS_NONE());
```

**Comment by lopopolo** [comment]
> The following C methods are not implemented.
> 
> ```c
> #ifdef MRB_ENABLE_ALL_SYMBOLS
>   mrb_define_class_method(mrb, s, "all_symbols", mrb_sym_all_symbols, MRB_ARGS_NONE());
> #endif
> ```

mruby-sys does not define this macro so this method is already not implemented.

File: artichoke-backend/src/extn/core/symbol/symbol.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+class Symbol
+  include Comparable
+
+  def capitalize
+    (to_s.capitalize! || self).to_sym
```
Comment (lopopolo) [inline artichoke-backend/src/extn/core/symbol/symbol.rb]: some of these implementations from mruby are a bit funky. For example, the docs for [`Symbol#capitalize`](https://ruby-doc.org/core-2.6.3/Symbol.html#method-i-capitalize) say the method is equivalent to:

```ruby
sym.to_s.capitalize.intern
```

I'd like to match the docs (even if the semantics are the same) if possible.

File: artichoke-backend/src/extn/core/symbol/symbol.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  def downcase
+    (to_s.downcase! || self).to_sym
```
Comment (lopopolo) [inline artichoke-backend/src/extn/core/symbol/symbol.rb]: docs say this is equivalent to

```ruby
sym.to_s.downcase.intern
```

File: artichoke-backend/src/extn/core/symbol/symbol.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  def casecmp?(sym)
+    c = casecmp(sym)
+    return nil if c.nil?
+
+    c.zero?
+  end
+
+  def empty?
+    to_s.empty?
```
Comment (lopopolo) [inline artichoke-backend/src/extn/core/symbol/symbol.rb]: docs say to check for equality with `:''`

**Pull request merged**